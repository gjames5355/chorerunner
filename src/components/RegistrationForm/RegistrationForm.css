.register-container {
  height: 100vh;
  overflow-y: auto;
}

.register-container p {
  color: white;
  margin: 0;
  padding-top: 100px;
}

.register-container h2 {
  color: rgb(255, 217, 0);
  font-size: 30px;
  text-shadow: -1px -1px rgb(105, 90, 0), 1.5px 1.5px rgb(134, 112, 0);
}

.registration {
  width: 80%;
  margin: auto;
}

div[role="alert"] {
  color: white;
  font-size: 1.1rem;
  font-weight: 500;
  margin-top: 0.5rem;
}

.formItem {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  margin: 15px 0;
}

div[role="alert"] {
  color: white;
  font-size: 1.1rem;
  font-weight: 500;
  margin-top: 0.5rem;
  font-family: 'Electrolize', sans-serif;
}

.registration input {
  width: 200px;
}

.registration label {
  font-family: 'Press Start 2P', cursive;
  color: rgb(255, 217, 0);
  padding-bottom: 10px;
}

.registration button {
  width: 190px;
  text-align: center;

  margin: 10px;
  font-family: 'Press Start 2P', cursive;
  font-size: 20px;
  padding: 7px;
  background-color: rgb(255, 217, 0);
  text-shadow: -1px -1px rgb(105, 90, 0), 1px 1px white;
  color: #725F00;
  -webkit-border-radius: 7px;
  -moz-border-radius: 7px;
  -o-border-radius: 7px;
  border-radius: 7px;
  box-shadow: 0 .3em rgb(134, 112, 0);
  cursor: pointer;
}

.formFooter {
  padding-bottom: 80px;
}