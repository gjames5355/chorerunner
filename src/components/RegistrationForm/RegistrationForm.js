import React, { Component } from 'react';
import { Input, Required, Label } from '../Form/Form';
import AuthApiService from '../../services/auth-api-service';
import Button from '../Button/Button';
import './RegistrationForm.css';

class RegistrationForm extends Component {
  static defaultProps = {
    onRegistrationSuccess: () => {},
  };

  state = {
    name: '',
    username: '',
    password: '',
    error: null,
    validateError: {
      nameError: '',
      usernameError: '',
    },
  };

  firstInput = React.createRef();

  onChangeHandle = e => {
    this.setState({
      [e.target.name]: e.target.value,
    });
  };

  validateForm = () => {
    let name = this.state.name.trim();
    let userName = this.state.username.trim();

    let usernameError = '';
    let nameError = '';

    //Validates the persons name

    if (name.length < 6) {
      nameError = 'Please enter more than 6 characters';
    }
    if (name.length > 50) {
      nameError = 'Your name must be less than 50 characters';
    }

    //Validates the username
    if (userName.length < 6) {
      usernameError = 'Please enter more than 6 characters';
    }
    if (userName.length > 50) {
      usernameError = 'Your name must be less than 50 characters';
    }

    if (usernameError || nameError) {
      this.setState({ validateError: { usernameError, nameError } });
      return false;
    }
    return true;
  };

  handleSubmit = ev => {
    ev.preventDefault();
    const isValid = this.validateForm();
    const formError = this.state.error;

    const { name, username, password } = ev.target;

    if (isValid) {
      AuthApiService.postUser({
        name: name.value,
        username: username.value,
        password: password.value,
      })
        .then(user => {
          name.value = '';
          username.value = '';
          password.value = '';
          this.props.onRegistrationSuccess();
        })
        .catch(res => {
          this.setState({ error: res.error });
        });
    }
    if (formError) {
      this.setState({
        name: '',
        username: '',
        password: '',
        error: null,
        validateError: {},
      });
    }
  };

  render() {
    const { name, username, password, error } = this.state;
    const { nameError, usernameError } = this.state.validateError;

    return (
      <div className='box'>
        <form
          className='registration'
          onSubmit={this.handleSubmit}
          name='registration-form'
        >
          <div className='formItem'>
            <Label htmlFor='registration-name-input'>
              Enter your name
              <Required />
            </Label>
            <Input
              className='formBox'
              ref={this.firstInput}
              id='registration-name-input'
              name='name'
              value={name}
              onChange={this.onChangeHandle}
              required
            />
            <div role='alert'>{nameError}</div>
          </div>
          <div className='formItem'>
            <Label htmlFor='registration-username-input'>
              Choose a username
              <Required />
            </Label>
            <Input
              className='formBox'
              id='registration-username-input'
              name='username'
              value={username}
              onChange={this.onChangeHandle}
              required
            />
            <div role='alert'>{usernameError}</div>
          </div>
          <div className='formItem'>
            <Label htmlFor='registration-password-input'>
              Choose a password
              <Required />
            </Label>
            <Input
              className='formBox'
              id='registration-password-input'
              name='password'
              type='password'
              value={password}
              onChange={this.onChangeHandle}
              required
            />
          </div>
          <div role='alert'>{error && <p className='alertMsg'>{error}</p>}</div>
          <footer className='formFooter'>
            <Button type='submit' className='basicBtn'>
              Sign up
            </Button>{' '}
            <br />
          </footer>
        </form>
      </div>
    );
  }
}

export default RegistrationForm;
