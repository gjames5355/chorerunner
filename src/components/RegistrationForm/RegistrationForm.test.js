import React from 'react';
import ReactDOM from 'react-dom';
import RegistrationForm from './RegistrationForm.js';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

it('renders ResultsPage without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <MemoryRouter>
      <RegistrationForm />
    </MemoryRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('renders ResultsPage as expected', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <RegistrationForm name='RegistrationForm' />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
