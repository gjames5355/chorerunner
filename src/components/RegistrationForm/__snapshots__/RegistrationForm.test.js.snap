// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ResultsPage as expected 1`] = `
<div
  className="box"
>
  <form
    className="registration"
    name="registration-form"
    onSubmit={[Function]}
  >
    <div
      className="formItem"
    >
      <label
        className="Label"
        htmlFor="registration-name-input"
      >
        Enter your name
        <span
          className="Required"
        >
          *
        </span>
      </label>
      <input
        className="Input formBox"
        id="registration-name-input"
        name="name"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
      <div
        role="alert"
      >
        
      </div>
    </div>
    <div
      className="formItem"
    >
      <label
        className="Label"
        htmlFor="registration-username-input"
      >
        Choose a username
        <span
          className="Required"
        >
          *
        </span>
      </label>
      <input
        className="Input formBox"
        id="registration-username-input"
        name="username"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
      <div
        role="alert"
      >
        
      </div>
    </div>
    <div
      className="formItem"
    >
      <label
        className="Label"
        htmlFor="registration-password-input"
      >
        Choose a password
        <span
          className="Required"
        >
          *
        </span>
      </label>
      <input
        className="Input formBox"
        id="registration-password-input"
        name="password"
        onChange={[Function]}
        required={true}
        type="password"
        value=""
      />
    </div>
    <div
      role="alert"
    />
    <footer
      className="formFooter"
    >
      <button
        className="Button basicBtn"
        type="submit"
      >
        Sign up
      </button>
       
      <br />
    </footer>
  </form>
</div>
`;
